src/test/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/integration/IncentiveLevelResourceTestBase.kt

```diff
@@ -76,27 +78,31 @@ class IncentiveLevelResourceTestBase : SqsIntegrationTestBase() {
     assertThat(queueSize).isEqualTo(0)
   }
 
-  protected fun assertDomainEventSent(eventType: String): HMPPSDomainEvent {
+  protected val anyDomainEvent = object : TypeReference<HMPPSDomainEvent<Any?>>() {}
+  protected val incentiveLevelDomainEvent = object : TypeReference<HMPPSDomainEvent<IncentiveLevelRef>>() {}
+  protected val prisonIncentiveLevelDomainEvent = object : TypeReference<HMPPSDomainEvent<PrisonIncentiveLevelRef>>() {}
+
+  protected fun <T> assertDomainEventSent(eventType: String, typeReference: TypeReference<HMPPSDomainEvent<T>>): HMPPSDomainEvent<T> {
     val sqsClient = incentivesQueue.sqsClient
     val queueSize = sqsClient.getApproxQueueSize(testDomainEventQueueUrl!!)
     assertThat(queueSize).isEqualTo(1)
 
     val body = sqsClient.receiveMessage(testDomainEventQueueUrl).messages[0].body
     val (message, attributes) = objectMapper.readValue(body, HMPPSMessage::class.java)
     assertThat(attributes.eventType.Value).isEqualTo(eventType)
-    val domainEvent = objectMapper.readValue(message, HMPPSDomainEvent::class.java)
+    val domainEvent = objectMapper.readValue(message, typeReference)
     assertThat(domainEvent.eventType).isEqualTo(eventType)
 
     return domainEvent
   }
 
-  protected fun getPublishedDomainEvents(): List<HMPPSDomainEvent> {
+  protected fun getPublishedDomainEvents(): List<HMPPSDomainEvent<Map<String, *>>> {
```

Comment:
`Map<String, *>` is an effective super set of `IncentiveLevelRef` and `PrisonIncentiveLevelRef`